pkg/action/build-disk.go

```diff
@@ -80,7 +80,18 @@ func NewBuildDiskAction(cfg *v1.BuildConfig, spec *v1.DiskSpec, opts ...BuildDis
 	}
 
 	if b.snapshotter == nil {
-		b.snapshotter, err = snapshotter.NewLoopDeviceSnapshotter(cfg.Config, cfg.Snapshotter, b.bootloader)
+		b.snapshotter, err = snapshotter.NewSnapshotter(cfg.Config, cfg.Snapshotter, b.bootloader)
+	}
+
+	if b.cfg.Snapshotter.Type == constants.BtrfsSnapshotterType {
+		if !b.spec.Expandable {
+			cfg.Logger.Errorf("Non expandable disk images are not supported for btrfs snapshotter")
```

Comment:
Shouldn't be hard to implement support for non expandable (full disk including all partitions) disk images in  way that is compatible for btrfs snapshotter too, but requires some build-disk command refactor which is beyond the scope of this PR and also I don't think is really a relevant use case for Elemental.